internal/pkg/manifest/svc.go

```diff
@@ -272,17 +273,49 @@ func (a *AdvancedCount) validScalingFields() []string {
 	}
 }
 
-func (a *AdvancedCount) hasScalingFieldsSet() bool {
-	switch a.workloadType {
-	case LoadBalancedWebServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case BackendServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case WorkerServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.QueueScaling.IsEmpty()
-	default:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty() || !a.QueueScaling.IsEmpty()
+func (a *AdvancedCount) getValidScalingFieldsSet() ([]string, error) {
+	scalingFieldNames := []string{"CPU", "Memory", "Requests", "ResponseTime", "QueueScaling"}
+	validFieldsSet := []string{}
+	invalidFieldsSet := []string{}
+
+	validTags := a.validScalingFields()
+	typ := reflect.ValueOf(*a).Type()
+
+	for _, name := range scalingFieldNames {
+		currStructField, ok := typ.FieldByName(name)
+		if ok {
+			zero := reflect.Zero(currStructField.Type).Interface()
+			value := reflect.ValueOf(a).Elem().FieldByName(name).Interface()
+			if zero != value {
+				tag := currStructField.Tag.Get("yaml")
+				isValid := false
+
+				for _, v := range validTags {
+					if tag == v {
+						validFieldsSet = append(validFieldsSet, tag)
+						isValid = true
+					}
+				}
+				if !isValid {
+					invalidFieldsSet = append(invalidFieldsSet, tag)
+				}
+			}
+		}
+	}
+
+	if len(invalidFieldsSet) > 0 {
+		return validFieldsSet,
+			&errInvalidAutoscalingFieldsWithWkldType{
+				invalidFields: invalidFieldsSet,
+				workloadType:  a.workloadType,
+			}
 	}
+	return validFieldsSet, nil
+}
+
+func (a *AdvancedCount) hasScalingFieldsSet() bool {
+	fields, _ := a.getValidScalingFieldsSet()
```

Comment:
We could do the error handling here.

Reply:
How would you propose to do this? Make hasScalingFieldsSet() return both a bool and an error?

Reply:
yeah because when `getValidScalingFieldsSet` throws an error, it doesn't make sense to proceed i guess 🤔

Reply:
Hm, I agree with you, but running it down the line, this change makes a lot of the cases where `hasScalingFieldSet()` and `hasAutoscaling()` clunkier to user. This is because they're both generally used in one line boolean statements, and returning two values would end up making these all a lot more complicated. It ends up adding lines to check this error in multiple places, whereas the current implementation just checks for the error once.

Reply:
Is there a way to maintain the easy to read one-line statements while ignoring the error check?